app/models/blueprint.rb

```diff
@@ -70,6 +74,17 @@ def content
     end
   end
 
+  def publish(bundle_name = nil)
+    publish_bundle.tap do
+      the_bundle = bundle
+      the_bundle.name = bundle_name if bundle_name
+      the_bundle.display = true # visible for ordering service
+      the_bundle.save!
+
+      update_attribute(:status, 'published')
```

Comment:
Use `update_attributes(:status => 'published')` because `update_attribute` skips validation.  http://apidock.com/rails/ActiveRecord/Persistence/update_attribute